### PR TITLE
Move most of the simd operators into an optional module

### PR DESCRIPTION
**Pull request by stephentyrone**
Adding simd to the stdlib caused some typechecker regressions. We can resolve them in the near-term by making the types universally available, but moving the arithmetic operators into a separate module that must be explicitly imported.

The working name for this module is `SIMDOperators`. We can tweak that going forward if we want to, but it seems generally plausible. I deliberately avoided `SIMD` to avoid collision with the existing SDK module named `simd`.

<rdar://problem/46364455>

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by stephentyrone** [comment]
@swift-ci please test

File: stdlib/public/core/SIMDVector.swift [hunk 1]
```diff
@@ -4,13 +4,6 @@ infix operator .< : ComparisonPrecedence
 infix operator .<= : ComparisonPrecedence
 infix operator .> : ComparisonPrecedence
 infix operator .>= : ComparisonPrecedence
-infix operator .& : LogicalConjunctionPrecedence
-infix operator .^ : LogicalDisjunctionPrecedence
-infix operator .| : LogicalDisjunctionPrecedence
-infix operator .&= : AssignmentPrecedence
-infix operator .^= : AssignmentPrecedence
-infix operator .|= : AssignmentPrecedence
-prefix operator .!
```
Comment (jrose-apple) [inline stdlib/public/core/SIMDVector.swift]: These should probably still be declared in the stdlib even if there are no functions that implement them.

**Comment by compnerd** [comment]
@stephentyrone - why not make this an overlay?  It could overlap the SIMD module in the SDK and re-export it with the additional overloads.

**Review by xedin — APPROVED** [review]
LGTM! (Concur with Jordan's note)

It's concerning to me that besides performance newly added operators re-introduced crashers, something has to be really fragile in the type-checker, going to take a closer look shortly.

**Comment by stephentyrone** [comment]
@compnerd the simd SDK module depends on having the operators available. I'm not sure why you want to reverse that dependency?

**Comment by stephentyrone** [comment]
@xedin Yeah, something is fishy about those two crashers. Clearly they weren't so much "fixed" as "happened to stop crashing" =)

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by xedin** [comment]
@stephentyrone One more thing I noticed might be good to do before merging - can you uncomment this expression https://github.com/apple/swift/blob/master/test/Sema/complex_expressions.swift#L7 and see if that works again?

**Comment by stephentyrone** [comment]
@xedin I'll put that in a separate PR, so we can get this moving (but yes, it does work again).

**Comment by xedin** [comment]
Okay, sounds good!

**Pull request merged**